apps/lex-core-service/src/main/java/com/infosys/lex/rating/service/RatingServiceImpl.java

```diff
@@ -134,23 +139,91 @@
+		ArrayList<Object> ratingInfo = new ArrayList<>();
+		for (Map<String, Object> ratingsInfo : ratingsInfoList) {
 			contentId = ratingsInfo.get("content_id").toString();
 			ratingsInfo.remove("content_id");
-			ratingsInfo.put("averageRating",df.format((float)ratingsInfo.get("averageRating") ));
+			ratingsInfo.put("averageRating", df.format((float) ratingsInfo.get("averageRating")));
+
+			List<Float> ratings = userResourceRatingRepo.getRatingsForContent(rootOrg, contentId);
+			Map<Float, Long> streamResult = ratings.stream().collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
+			for (Integer defaultValue : defaultRating) {
+				ratingValue = new HashMap<>(2);
+				ratingValue.put("rating", defaultValue);
+				ratingValue.put("count", streamResult.getOrDefault(Float.valueOf(defaultValue), 0L));
+				ratingInfo.add(ratingValue);
+			}
+			ratingsInfo.put("ratingInfo", ratingInfo);
+
 			resp.put(contentId, ratingsInfo);
 		}
 		return resp;
-		
+
+	}
+
+	/**
+	 *
+	 * @param rootOrg root org
+	 * @param contentId content Id
+	 * @return return rating info for a content Id
+	 */
+	@Override
+	public Map<String, Object> getRatingsInfoForContents(String rootOrg, String contentId) {
+		ArrayList<Object> ratingInfo = new ArrayList<>();
+		List<String> contentIds = new ArrayList<>(Collections.singletonList(contentId));
+		List<Map<String, Object>> ratingsInfoList = userResourceRatingRepo.getAvgRatingAndRatingCountForContentIds(rootOrg, contentIds);
+		Map<String, Object> resp = new HashMap<>();
+		List<Integer> defaultRating = new ArrayList<>(Arrays.asList(1, 2, 3, 4, 5));
+		if (!CollectionUtils.isEmpty(ratingsInfoList)) {
+			HashMap<String, Object> ratingValue;
+			Map<String, Object> ratingResponse = ratingsInfoList.get(0);
+			resp.put("averageRating", df.format((float) ratingResponse.get("averageRating")));
+			resp.put("ratingCount", ratingResponse.get("ratingCount"));
+
+			List<Float> ratings = userResourceRatingRepo.getRatingsForContent(rootOrg, contentId);
+			Map<Float, Long> streamResult = ratings.stream().collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
+			for (Integer defaultValue : defaultRating) {
+				ratingValue = new HashMap<>(2);
+				ratingValue.put("rating", defaultValue);
+				ratingValue.put("count", streamResult.getOrDefault(Float.valueOf(defaultValue), 0L));
+				ratingInfo.add(ratingValue);
+			}
+			resp.put("ratingInfo", ratingInfo);
+		}
+		return resp;
+	}
+
+	/**
+	 *
+	 * @param rootOrg
+	 * @param ratingSearchDTO
+	 * @return
+	 */
+	@Override
+	public List<Object> getAllRatingsForContent(String rootOrg, RatingSearchDTO ratingSearchDTO) {
+		Integer limit = ratingSearchDTO.getPageSize() == null ? defaultLimit : ratingSearchDTO.getPageSize();
+		List<Object> responseArray = new ArrayList<>();
+		HashMap<String, Object> response;
+		List<Map<String, Object>> ratings = userResourceRatingRepo.getRatingInfoForContent(rootOrg, ratingSearchDTO.getContentId(), limit);
```

Comment:
Handling of pagination is wrong here... The purpose of API is to get the ratings of all the contents which is displayed in particular page... 
The pagination is already handled when fetching a list of contents to display in a given page.... So no need of this API to fetch the ratings... 
We just need to get all the ratings for the given list which is already implemented above.